src/guide/agent-functions/defining-agent-functions.rst

```diff
@@ -74,12 +84,128 @@ When using the Run-Time Compiled (RTC) functions, optionally in the C++ API or r
     
 .. note::
 
-    If you wish to store RTC agent functions in separate files :func:`newRTCFunction()<flamegpu::AgentDescription::newRTCFunction>` can be replaced with :func:`newRTCFunctionFile()<flamegpu::AgentDescription::newRTCFunctionFile>`, instead passing the path to the agent function's source file (relative to the working directory at runtime). This will allow them to be developed in a text editor with C++/CUDA syntax highlighting.
+    If you wish to store RTC agent functions in separate files :func:`newRTCFunction()<flamegpu::AgentDescription::newRTCFunction>` can be replaced with :func:`newRTCFunctionFile()<flamegpu::AgentDescription::newRTCFunctionFile>`, instead passing the path to the agent function's source file (relative to the working directory at runtime). This will allow them to be developed in a text editor with C++ syntax highlighting.
+
+.. _Pure Python Agent Functions:
+
+FLAME GPU Pure Python Agent Functions
+-------------------------------------
+
+Ane experimental feature allows the specification of *pure* Python agent functions. In this case agent functions can be expressed using a limited subset of Python. This is limited to syntax which can be transpiled to a C++ function at runtime for compilation. For example Python functionality like generator expressions, arrays, dictionaries, etc, which do not have an obvious C++ equivalent (excluding the in the SDL which is not supported in agent functions) is not permitted and will raise an exception. 
+
+Pure Python agent functions are required to have the ``@pyflamegpu.agent_function`` decorator and must specify two arguments, the message in message out variable, both of which must use a type annotation of a :ref:`supported message type<Communication Strategies>` prefixed with the Python FLAME GPU module name ``pyflamegpu.``.
```

Comment:
> the message in message out variable

the `message_in` and `message_out` variables

Reply:
fixed